### PR TITLE
Remove duplicate error messages in account and address creation

### LINKED ISSUE #27876: FO - Create an account - the alert is duplicated when we create a new account with invalid firstname and lastname
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

In the FO > when we create an account from the `Create an account` page or checkout page with invalid `firstname` and `lastname`, the alert is duplicated
**Create account page**
image
**Checkout page**
image

In the PS1783 => OK

### Expected behavior

image

### Steps to reproduce

1. Go to FO > Create a new account
2. Set invalid firstname & lastname for example demo_
3. Set required fields and save 

-----------------------------------------

1. Go to FO > Add a product to cart & proceed to checkout
2. In the personal information step > Set invalid firstname & lastname for example demo_
3. Set required fields and save

### PrestaShop version(s) where the bug happened

1.7.8.4, develop

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | This PR is a revert of the PR #24600 as the "real" fix was #27376 that as been merged later. When keeping both, the validation happens twice leading to a duplicate of the error messages.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27876
| How to test?      | Please see #27876 & #24464
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28028)

### CODE REVIEW

File: classes/form/CustomerAddressForm.php [hunk 1]
```diff
@@ -104,7 +104,21 @@ public function fillWith(array $params = [])
...
+                if (!$country->checkZipCode($postcode->getValue())) {
+                    $postcode->addError($this->translator->trans(
+                        'Invalid postcode - should look like "%zipcode%"',
+                        ['%zipcode%' => $country->zip_code_format],
+                        'Shop.Forms.Errors'
+                    ));
+                    $is_valid = false;
+                }
+            }
+        }
```
Comment (Progi1984) [inline classes/form/CustomerAddressForm.php]:
```suggestion
        $postcode = $this->getField('postcode');
        if ($postcode && $postcode->isRequired()) {
            $country = $this->formatter->getCountry();
            if (!$country->checkZipCode($postcode->getValue())) {
               $postcode->addError($this->translator->trans(
                    'Invalid postcode - should look like "%zipcode%"',
                    ['%zipcode%' => $country->zip_code_format],
                    'Shop.Forms.Errors'
               ));
               $is_valid = false;
           }
        }
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

**Review by matthieu-rolland — DISMISSED** [review]
LGTM, there's a small feedback from @Progi1984

**Review by matks — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @atomiix, 

Thank you for your PR, I tested it and that works 

as you can see : 

On the creation of account 

https://user-images.githubusercontent.com/100699445/161756769-38498dab-6bd2-4c93-8fca-2b48f7440b53.mp4

After an order : 

https://user-images.githubusercontent.com/100699445/161756981-d653b949-1947-4f5e-b18e-ca25082d2932.mp4

on the adress location : 

https://user-images.githubusercontent.com/100699445/161757105-32aeac7c-627b-4ebf-9125-686281a6cc84.mp4

PS : for the last video we can think it's an issue on the "adress" field but these charater can be in some adresses

**Comment by kpodemski** [comment]
Thank you @atomiix @AureRita

**Pull request merged**